Makefile

```diff
@@ -123,11 +123,18 @@ mcphost: ## Attach to mcphost
 	@echo "Attaching to mcphost..."
 	./scripts/mcphost.sh
 
+.ONESHELL:
 test-eval: ## Run agent evaluation tests
-	@echo "Refreshing OCM token..."
-	@. utils/ocm-token.sh && get_ocm_token && echo "$$OCM_TOKEN" > test/evals/ocm_token.txt
-	@echo "Running agent evaluation tests..."
-	@cd test/evals && python eval.py
+	export UNIQUE_ID=$(head /dev/urandom | tr -dc 0-9a-z | head -c 8)
+	export TEMP_DIR=$(mktemp -d)
+	echo "Refreshing OCM token..."
+	. utils/ocm-token.sh && get_ocm_token && echo "$$OCM_TOKEN" > test/evals/ocm_token.txt
+	echo "Running agent evaluation tests..."
+	cp test/evals/eval_data.yaml $$TEMP_DIR/eval_data.yaml
+	echo "Replacing the placeholder 'uniq-cluster-name' with '$${UNIQUE_ID}' in eval_data.yaml"
+	sed -i "s/uniq-cluster-name/$${UNIQUE_ID}/g" $$TEMP_DIR/eval_data.yaml
+	cd test/evals && python eval.py --eval_data_yaml $$TEMP_DIR/eval_data.yaml
+	rm -Rf $$TEMP_DIR
```

Comment:
Can we maybe do this with some kind of `trap` in case the eval returns an error (and fails the target)?

Or is that not how this will work?